utils/hooks/useForm.ts

```diff
@@ -0,0 +1,58 @@
+import { useState, useReducer, ChangeEventHandler, FormEvent } from 'react'
+import { RESET, UPDATE, IMPORT } from '../constants'
+
+interface ActionState {
+    type: string
+    payload?: any
+}
+
+function reducer(state: any, action: ActionState) {
+    switch (action.type) {
+        case UPDATE: {
+            const { name, value } = action.payload
+            return { ...state, [name]: { value } }
+        }
+        case RESET: {
+            return action.payload
+        }
+        default:
+            return state
+    }
+}
+
+const useForm = (initialState: any) => {
+    const [form, dispatch] = useReducer(reducer, initialState)
+
+    const handleChange: ChangeEventHandler<HTMLInputElement> = event => {
+        let { value, name } = event.target
```

Comment:
Literally never use `let` unless you absolutely need to re-assign a value to the variable. `const` here